src/PaymentGateways/PayPalStandard/Webhooks/WebhookValidator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Give\PaymentGateways\PayPalStandard\Webhooks;
+
+use Give\Log\Log;
+
+/**
+ * This class use to validate PayPal Standard ipn.
+ * Validate the IPN: https://developer.paypal.com/docs/api-basics/notifications/ipn/IPNImplementation/
+ *
+ * @unreleased
+ */
+class WebhookValidator
+{
+    /**
+     * @unreleased
+     *
+     * @param array $eventData PayPal ipn body data.
+     *
+     * @return bool
+     */
+    public function verifyEventSignature(array $eventData)
+    {
+        $eventData = array_merge( [ 'cmd' => '_notify-validate' ], $eventData );
+
+        // Validate IPN request w/ PayPal if user hasn't disabled this security measure.
+        if (give_is_setting_enabled(give_get_option('paypal_verification', 'enabled'))) {
+            $requestArgs = [
+                'method' => 'POST',
+                'timeout' => 45,
+                'redirection' => 5,
+                'httpversion' => '1.1',
+                'blocking' => true,
+                'headers' => [
+                    'host' => give_is_test_mode() ? 'www.sandbox.paypal.com' : 'www.paypal.com',
+                    'connection' => 'close',
+                    'content-type' => 'application/x-www-form-urlencoded',
+                    'post' => '/cgi-bin/webscr HTTP/1.1',
+                ],
+                'sslverify' => false,
+                'body' => $eventData,
+            ];
+
+            $apiResponse = wp_remote_post(give_get_paypal_redirect(), $requestArgs);
+
+            if (is_wp_error($apiResponse)) {
+                Log::error(
+                    'PayPal Standard IPN Error',
+                    ['IPN Data' => $apiResponse]
+                );
+
+                return false;
+            }
+
+            if ('VERIFIED' !== $apiResponse['body']) {
+                Log::error(
+                    'PayPal Standard IPN Error',
+                    ['IPN Data' => $apiResponse]
+                );
+
+                return false;
+            }
+        }
+
+        return true;
```

Comment:
I would suggest moving this to a early return. Currently, the `if` check wraps most of the code, which creates a separation between the check and the fallback return. Instead, it would be more readable to invert the check and return early, which removes the nested code structure.

```php
function foo() {
    if( ! $settingEnabled ) {
        return true;
    }

    // Most of the code goes here
}
```

Reply:
fixed: https://github.com/impress-org/givewp/compare/29d95f2aadff...f961a2a7a6a2